runtime-sdk/modules/contracts/src/store.rs

```diff
@@ -12,7 +13,18 @@ pub fn for_instance<'a, C: Context>(
     ctx: &'a mut C,
     instance_info: &types::Instance,
     store_kind: StoreKind,
-) -> Result<impl Store + 'a, Error> {
+) -> Result<Box<dyn Store + 'a>, Error> {
+    let key_pair: Option<KeyPair> = if let StoreKind::Confidential = store_kind {
+        let kmgr_client = ctx.key_manager().ok_or(Error::Unsupported)?;
+        let kid = keymanager::get_key_pair_id(&[&instance_info.id.to_storage_key()]);
```

Comment:
This shouldn't directly use the instance ID, it should be namespaced (use some constant base context as the first element).